### PR TITLE
Pre-main jobs for reconciler PR with cli:main

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Added `pre-main-reconciler-cli-integration-kyma-latest-k3d` job. ([Test run](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-incubator_reconciler/899/mfaizanse_test_of_prowjob_pre-main-reconciler-cli-integration-k/1501902051863433216))
- Added `pre-main-reconciler-cli-integration-kyma-main-k3d` job. ([Test run](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-incubator_reconciler/899/mfaizanse_test_of_prowjob_pre-main-reconciler-cli-integration-k/1501906146347192320))

**Related issue(s)**

- https://github.com/kyma-incubator/reconciler/issues/807

### CODE REVIEW

File: prow/scripts/cluster-integration/reconciler-integration-with-cli-k3d.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  command -v node >/dev/null 2>&1 || { echo >&2 "node not found"; exit 1; }
+  command -v npm >/dev/null 2>&1 || { echo >&2 "npm not found"; exit 1; }
+  command -v jq >/dev/null 2>&1 || { echo >&2 "jq not found"; exit 1; }
+  command -v helm >/dev/null 2>&1 || { echo >&2 "helm not found"; exit 1; }
+  command -v kubectl >/dev/null 2>&1 || { echo >&2 "kubectl not found"; exit 1; }
+  command -v k3d >/dev/null 2>&1 || { echo >&2 "k3d not found"; exit 1; }
+}
+
+function load_env() {
+  ENV_FILE=".env"
```
Comment (nachtmaar) [inline prow/scripts/cluster-integration/reconciler-integration-with-cli-k3d.sh:18]: // TODO(nils): What is inside .env ?

File: templates/data/incubator-buildpack-data.yaml [hunk 2]
```diff
@@ -344,6 +348,52 @@ templates:
...
+                  run_if_changed: "^((cmd\\S+|configs\\S+|internal\\S+|pkg\\S+)(\\.[^.][^.][^.]+$|\\.[^.][^dD]$|\\.[^mM][^.]$|\\.[^.]$|/[^.]+$))"
+                  labels:
+                    preset-build-pr: "true"
+                    preset-kyma-guard-bot-github-token: "true"
+                    preset-sa-vm-kyma-integration: "true"
+                    preset-gc-project-env: "true"
+                  annotations:
+                    pipeline.platform: k3d
+                  env:
+                    KYMA_TEST_SOURCE: "latest-release"
```
Comment (nachtmaar) [inline templates/data/incubator-buildpack-data.yaml:363]: Where is thing variable being used ? 🤔 I see that setting the version works (in the logs). Can you explain it ?

**Review by nachtmaar — APPROVED** [review]

File: templates/data/incubator-buildpack-data.yaml (hunk 2 above)
Comment (mfaizanse) [reply templates/data/incubator-buildpack-data.yaml:363]: Its used [here](https://github.com/kyma-project/test-infra/blob/92a1fe680720546d298182b52526dd550f8b12c0/prow/scripts/provision-vm-and-test-reconciler-with-cli-k3d.sh#L124)!

**Pull request merged**